imap_processing/lo/l1a/lo_l1a.py

```diff
@@ -29,156 +29,139 @@ def lo_l1a(dependency: Path, data_version: str) -> list[Path]:
+        azimuth_6 = xr.DataArray(
+            data=np.arange(0, 60, dtype=np.uint8),
+            name="azimuth_6",
+            dims=["azimuth_6"],
+            attrs=attr_mgr.get_variable_attributes("azimuth_6"),
+        )
+        azimuth_6_label = xr.DataArray(
+            data=azimuth_6.values.astype(str),
+            name="azimuth_6_label",
+            dims=["azimuth_6_label"],
+            attrs=attr_mgr.get_variable_attributes("azimuth_6_label"),
         )
 
-        dataset = xr.Dataset(
-            coords={"epoch": epoch_time, "direct_events": direct_events},
-            attrs=attr_mgr.get_global_attributes(logical_source),
+        esa_step = xr.DataArray(
+            data=np.arange(1, 8, dtype=np.uint8),
+            name="esa_step",
+            dims=["esa_step"],
+            attrs=attr_mgr.get_variable_attributes("esa_step"),
+        )
+        esa_step_label = xr.DataArray(
+            esa_step.values.astype(str),
+            name="esa_step_label",
+            dims=["esa_step_label"],
+            attrs=attr_mgr.get_variable_attributes("esa_step_label"),
         )
 
-    # Loop through the data fields that were pulled from the
-    # data class. These should match the field names given
-    # to each field in the YAML attribute file
-    for data_field in data_fields:
-        field = data_field.name.lower()
-        # TODO: should add ground_sw_version and packet_file_name to
-        # CDF and should use IntVar for DATA. Should only need to skip
-        # SHCOARSE and ccsds_header (unless Lo wants this in the CDF as well).
-        if field not in [
-            "shcoarse",
-            "ground_sw_version",
-            "packet_file_name",
-            "ccsds_header",
-            "data",
-        ]:
-            # Create a list of all the dimensions using the DEPEND_I keys in the
-            # YAML attributes
-            dims = [
-                value
-                for key, value in attr_mgr.get_variable_attributes(field).items()
-                if "DEPEND" in key
-            ]
+        dataset.shcoarse.attrs.update(attr_mgr.get_variable_attributes("shcoarse"))
+        dataset.epoch.attrs.update(attr_mgr.get_variable_attributes("epoch"))
 
-            # Create a data array for the current field and add it to the dataset
-            # TODO: need to update to use packet data once that's available
-            dataset[field] = xr.DataArray(
-                [1, 1, 1], dims=dims, attrs=attr_mgr.get_variable_attributes(field)
-            )
+        dataset = dataset.assign_coords(
+            azimuth_60=azimuth_60,
+            azimuth_60_label=azimuth_60_label,
+            azimuth_6=azimuth_6,
+            azimuth_6_label=azimuth_6_label,
+            esa_step=esa_step,
+            esa_step_label=esa_step_label,
+        )
+        dataset.attrs.update(attr_mgr.get_global_attributes(logical_source))
+        # remove the binary field and CCSDS header from the dataset
```

Comment:
Would it be helpful to add a flag to remove the header from the packet_file_to_datasets return? I think @tech3371 suggested that as well, so maybe this is an indication it would be nice for multiple people.

Reply:
Yes, I agree I think having that flag would be helpful.